osquery/tables/events/linux/seccomp_events.cpp

```diff
@@ -0,0 +1,523 @@
+  subscribe(&SeccompEventSubscriber::Callback, sc);
+
+  return Status();
+}
+
+Status SeccompEventSubscriber::Callback(const ECRef& ec, const SCRef& sc) {
+  QueryData data;
+  auto status = processEvents(data, ec->audit_events);
+  if (!status.ok()) {
+    return status;
+  }
+
+  for (auto& row : data) {
+    add(row);
+  }
+
+  return Status();
+}
+
+void SeccompEventSubscriber::parseEvent(const AuditEvent& event,
+                                        Row& parsed_event) noexcept {
+  const auto& event_data = boost::get<SeccompAuditEventData>(event.data);
+  // parsed_event["uptime"] = std::to_string(tables::getUptime());
+  parsed_event["uptime"] = std::to_string(getUptime());
+  std::uint64_t arch = 0;
+  std::uint64_t value = 0;
+
+  for (const auto& field : event_data.fields) {
+    switch (field.second.which()) {
+    case 0:
+      // string field
+      parsed_event[field.first] = boost::get<std::string>(field.second);
+      break;
+    case 1:
+      // int field
+      value = boost::get<const std::uint64_t>(field.second);
+      if (field.first == "arch") {
+        std::string symbolic_code_value = "";
+        arch = value;
+        parsed_event[field.first] =
+            GetStringFieldFromIntMap(symbolic_code_value,
+                                     SeccompEventSubscriber::arch_codes_map,
+                                     value)
+                ? symbolic_code_value
+                : std::to_string(value) + "(unknown)";
+        break;
+      }
+      if (field.first == "syscall" && arch == AUDIT_ARCH_X86_64) {
+        std::string symbolic_code_value = "";
+        parsed_event[field.first] =
+            GetStringFieldFromIntMap(symbolic_code_value,
+                                     SeccompEventSubscriber::syscall_x86_64_map,
+                                     value)
+                ? symbolic_code_value
+                : std::to_string(value) + "(unknown)";
+        break;
+      }
+      if (field.first == "syscall" && arch != AUDIT_ARCH_X86_64) {
+        parsed_event[field.first] = std::to_string(value);
+        break;
+      }
+      if (field.first == "code") {
+        std::string symbolic_code_value = "";
+        parsed_event[field.first] =
+            GetStringFieldFromIntMap(
+                symbolic_code_value,
+                SeccompEventSubscriber::seccomp_actions_map,
+                value)
+                ? symbolic_code_value
+                : std::to_string(value) + "(unknown)";
+        break;
+      }
+      parsed_event[field.first] = UNSIGNED_BIGINT(value);
+      break;
+    default:
+      continue;
+    }
+  }
+}
+
+Status SeccompEventSubscriber::processEvents(
+    QueryData& data, const std::vector<AuditEvent>& event_list) noexcept {
+  data.clear();
+
+  for (const auto& event : event_list) {
+    if (event.type != AuditEvent::Type::Seccomp) {
+      continue;
+    }
+
+    if (event.record_list.size() != 1) {
+      VLOG(1) << "SeccompEventSubscriber got record list";
```

Comment:
I am curious about this `VLOG`, does an empty record list or a record list with more than one value mean something different?

Reply:
That part was written a year ago and mostly inspired by apparmor_events table patch.
Not sure about this VLOG, most propably that was also taken from apparmor_events. According to event [processing](https://github.com/osquery/osquery/pull/7006/files#diff-6c8633417878b99e2265f7f785e2cdbf08dfe1e0b0d21fa345ccdfdd9ec989d0R195-R196) here should be only one entry in list, and this is just semiquiet assert